telemetry/client.go

```diff
@@ -60,74 +66,70 @@ type Client struct {
 	keyUID             string
 	nodeName           string
 	version            string
-	receivedMessagesCh chan ReceivedMessages
-	receivedEnvelopeCh chan *v2protocol.Envelope
-	sentEnvelopeCh     chan wakuv2.SentEnvelope
 	telemetryCh        chan TelemetryRequest
+	telemetryCacheLock sync.Mutex
+	telemetryCache     []TelemetryRequest
+	nextIdLock         sync.Mutex
 	nextId             int
 	sendPeriod         time.Duration
 }
 
-func NewClient(logger *zap.Logger, serverURL string, keyUID string, nodeName string, version string) *Client {
+func NewClient(logger *zap.Logger, serverURL string, keyUID string, nodeName string, version string, sendPeriod ...time.Duration) *Client {
```

Comment:
I am not sure if we need to make the sendPeriod optional - you could even leave it as is just without the `...` and then check for `0` and set default.

Reply:
I only needed this for testing, so I figured making it optional saves having to change the call to `NewClient` in all other places in the code

Reply:
When dogfooding these changes in status-desktop, this actually ended up causing an issue because I guess you can still access `[0]` on a slice even if it's `nil`. I replaced it with the pattern used more commonly in Go (and in status-go) which is functional options.